indexing-service/src/main/java/org/apache/druid/indexing/overlord/SingleTaskBackgroundRunner.java

```diff
@@ -314,33 +317,33 @@ public TaskLocation getTaskLocation(String taskId)
   }
 
   @Override
-  public long getTotalTaskSlotCount()
+  public Map<String, Long> getTotalTaskSlotCount()
   {
-    return 1;
+    return ImmutableMap.of(WorkerConfig.DEFAULT_CATEGORY, 1L);
```

Comment:
AFAIT, the returned value is currently not used in anywhere. @mghosh4 is this correct? If so, how about throwing `UnsupportedOperationException` instead to make it more obvious?

Reply:
@jihoonson yes you are correct this value is not used anywhere except `SingleTaskBackgroundRunnerTest`. Returning 1 is probably the right thing to do here since in a sense there is only 1 task slot. Ofcourse throwing the exception is an option as well. I would personally tilt towards the first option given it fits the semantics better.

Reply:
I would agree that returning 1 is reasonable if there was no category returned in this method as it was before. Now, this method always returns the default category, which could be wrong when your middleManager has a different category configured. It seems a bit clearer to me if it throws `UnsupportedOperationException`. In any case, it would be great if you can add some comment saying "This method should be never called in peons".